app/src/main/java/de/xikolo/controllers/helper/VideoSettingsHelper.kt

```diff
@@ -0,0 +1,243 @@
+package de.xikolo.controllers.helper
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.provider.Settings
+import android.support.annotation.StringRes
+import android.support.v4.content.ContextCompat
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import android.widget.TextView
+import de.xikolo.R
+import de.xikolo.models.VideoSubtitles
+import de.xikolo.utils.PlaybackSpeedUtil
+import java.util.*
+
+class VideoSettingsHelper(private val context: Context, private val subtitles: List<VideoSubtitles>?, private val changeListener: OnSettingsChangeListener, private val clickListener: OnSettingsClickListener, private val qualityOfflineInfo: QualityOfflineInfo) {
+
+    enum class VideoMode {
+        SD, HD
+    }
+
+    private val inflater: LayoutInflater = LayoutInflater.from(context)
+
+    var currentQuality: VideoMode = VideoMode.HD
+    var currentSpeed: PlaybackSpeedUtil = PlaybackSpeedUtil.x10
+    var currentVideoSubtitles: VideoSubtitles? = null
+
+    fun buildSettingsView(): ViewGroup {
+        val list = buildSettingsPanel(null)
+
+        list.addView(
+            buildSettingsItem(
+                R.string.icon_quality,
+                context.getString(R.string.video_settings_quality) + "  " + context.getString(R.string.video_settings_separator) + "  " + currentQuality.toString() +
+                    if (qualityOfflineInfo.isOfflineAvailable(currentQuality)) " " + context.getString(R.string.video_settings_quality_offline) else "",
```

Comment:
Is the `else` required?

Reply:
Sadly yes, because `if` requires the `else` branch if used in an expression: [see here](http://kotlinlang.org/docs/reference/control-flow.html#if-expression)